### PR TITLE
Bounty rendering subcomponents inside of it. 

### PR DESCRIPTION
**Pull request by Skarrg**
Uncomplicated the file structure for Bounty Component

Renders Basic Bounty area in it.

### CODE REVIEW

**Comment by DChan0319** [comment]
Make sure you assign Someone to review the code. Also why are you merging from a component straight onto master. merge flow should be as follows, whatever task/component/functionality -> dev -> master not task/component/functionality -> master.

File: src/Components/Bounty/index.js [hunk 1]
```diff
@@ -1,10 +1,66 @@
...
-);
+const BountyInfo = () => null;
+const BountyReward = () => null;
+const BountyTitle = () => null;
 
+class Bounty extends Component {
+
+    renderTitle() {
+        const { children } = this.props;
+        const bountyTitle = findByType(children, BountyTitle);
```
Comment (DChan0319) [inline src/Components/Bounty/index.js:15]: Are you searching the DOM for an element and then storing that as the bountyTitle??

File: src/Components/Bounty/index.js [hunk 2]
```diff
@@ -1,10 +1,66 @@
...
+                </div>
+                <div className="bountyRewardContainer">
+                    {this.renderReward()}
+                </div>
+            </div>
+        )
+    }
+}
+
+Bounty.BountyTitle = BountyTitle;
```
Comment (DChan0319) [inline src/Components/Bounty/index.js:63]: Why are you adding properties to this react component like this? You don't want to add properties to a react component. This is not common practice. Instead, they should probably be states

File: src/App.js [hunk 3]
```diff
@@ -1,26 +1,16 @@
...
-            href="https://reactjs.org"
-            target="_blank"
-            rel="noopener noreferrer"
-          >
-            Learn React
-          </a>
-        </header>
-      </div>
+      <Bounty>
+        <Bounty.BountyTitle></Bounty.BountyTitle>
```
Comment (DChan0319) [inline src/App.js:10]: I wouldnt render these components like this in your app. what does your bounty contain? a title, info, and reward? Your bounty component should contain these components. and the app should only render the bounty. bounty/index.js should then contain all of your other components which you have in your divs. I dont see why you need to attach these elements to the app. Looks repetitive.

File: src/Components/Bounty/index.js [hunk 4]
```diff
@@ -1,10 +1,66 @@
...
+        }
+
+        return <div className="bountyReward">BountyReward Rendered! {bountyReward.props.children}</div>
+    }
+
+    render() {
+        return (
+            <div className="bountyContainer">Bounty Rendered!
+                <div className="bountyTitleContainer">
+                    {this.renderTitle()}
```
Comment (DChan0319) [inline src/Components/Bounty/index.js:50]: this is where you render your bounty component with its other components inside. why do you need to have <bounty.[subComponent]> inside app.js?

File: src/utils/findByType.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+        const childType =
+            child && child.type && (child.type.displayName || child.type.name);
+        if (type.includes(childType)) {
+            result.push(child);
+        }
+    });
+
+    return result[0];
+};
+export default findByType;
```
Comment (DChan0319) [inline src/utils/findByType.js:18]: add an empty line at end of file.

File: src/Components/Bounty/style.css [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+    background-color: lightblue
+}
+
+.bountyInfo {
+    background-color: green
+}
+
+
+.bountyReward{
+    background-color: gold
```
Comment (DChan0319) [inline src/Components/Bounty/style.css:15]: Use color hex and pull these colors out to css variables. You can pick your colors with a google chrome extension called color picker or w/e. Have a main.css and import all your other styles into your main.css you can put these color variables into the variables.css file and then import that into your main.css and in your index.html all you do is link up the main.css contact me for more clarification

File: src/utils/findByType.js [hunk 7]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+
+const findByType = (children, component) => {
```
Comment (DChan0319) [inline src/utils/findByType.js:3]: why do you need to find children components??

**Review by DChan0319 — CHANGES_REQUESTED** [review]

**Comment by jmckenney** [comment]
The green build on this repo is 🥇

**Pull request closed without merging**